Content.Server/GameObjects/Components/Stack/StackComponent.cs

```diff
@@ -23,9 +24,34 @@ public class StackComponent : SharedStackComponent, IInteractUsing, IExamine
         public override int Count
         {
             get => base.Count;
-            set => base.Count = value;
+            set
+            {
+                base.Count = value;
+                
+                if (base.Count != 0 && SpriteStates != null && Owner.TryGetComponent<SpriteComponent>(out var spriteComponent))
```

Comment:
remove the `SpriteStates != null` check (related to other comment in `ExposeData`)

Reply:
don't, as discussed above